File: 0219_contains_duplicate_II.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Boolean}
def contains_nearby_duplicate(nums, k)
    map = {}

    nums.each_with_index do |num, i|
        if map.key?(num) 
            return true if (map[num] - i).abs <= k
        end
        map[num] = i
    end
    false
end

File: 0219_contains_duplicate_II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    const map = new Map();

    for (let i = 0; i < nums.length; ++i) {
        if(!map.has(nums[i])) {
            map.set(nums[i], i);
        } else {
            if (Math.abs(map.get(nums[i]) - i) <= k) {
                return true;
            } else {
                map.set(nums[i], i);
            }
        }
    }
    return false;
};
